pulseapi/profiles/serializers.py

```diff
@@ -45,7 +45,10 @@ class UserProfileSerializer(serializers.ModelSerializer):
 
     def __init__(self, instance=None, *args, **kwargs):
         super().__init__(instance, *args, **kwargs)
-        if instance is not None:
+
+        if instance is not None and type(instance) is UserProfile:
+            # We type-check to prevent this from kicking in for entire
```

Comment:
I don't follow this. It would kick in with the result of the queryset object only if it is a UserProfile without this check wouldn't it?

Reply:
tell that to Django: without the typecheck the tests died because `instance` turned out to be an entire QuerySet instead of a single object =/

Reply:
oh ok this makes sense since drf dynamically decides whether to implement ListSerializer methods or Serializer methods based on the `many=True` argument passed when creating an instance of the serializer.